release: v0.2.0

ymjiang opened the pull request:
This PR is for the official release of `v0.2.0`

byronyi commented:
Congrats! It will be better to add a `CHANGELOG.rst` file to mention the important changes from last version.

ymjiang commented:
Added a changelog to indicate the major changes. @byronyi

ymjiang commented:
@bobzhuyb Please take a review. Major changes:
- Uploaded pip package: `pip3 install byteps`
- Updated all docker images.
- Fixed stale documents.

bobzhuyb commented on CHANGELOG.rst:
```diff
@@ -0,0 +1,16 @@
+
+0.2.0 (2020-02)
+------------------
+* Largely improve RDMA performance by enforcing page aligned memory and adding IPC support. Now support colocating servers and workers without sacrificing much performance.
```
page aligned and IPC support can be two separate items.

bobzhuyb commented on CHANGELOG.rst:
```diff
@@ -0,0 +1,16 @@
+------------------
+* Largely improve RDMA performance by enforcing page aligned memory and adding IPC support. Now support colocating servers and workers without sacrificing much performance.
+* Fix a hanging bug in BytePS server.
+* Fix RDMA fork problem caused by multi-processing (e.g., data loaders).
```
Fix RDMA-related segmentation fault problem during fork() (e.g., used by PyTorch data loader).

bobzhuyb commented on CHANGELOG.rst:
```diff
@@ -0,0 +1,16 @@
+* Fix a hanging bug in BytePS server.
+* Fix RDMA fork problem caused by multi-processing (e.g., data loaders).
+* New feature: Enable mixing use of colocate and non-colocate servers, along with a smart tensor allocation strategy.
+* Add support for pip install.
```
Add the pip install command here.

bobzhuyb commented on README.md:
```diff
@@ -41,21 +46,30 @@ BytePS also incorporates many acceleration techniques such as hierarchical strat
-Below, we explain how to build and run BytePS by yourself. BytePS assumes that you have already installed one or more of the following frameworks: TensorFlow / PyTorch / MXNet. BytePS depends on CUDA and NCCL, and requires gcc>=4.9. If you are working on CentOS/Redhat and have gcc<4.9, you can try `yum install devtoolset-7` before everything else.
+Below, we explain how to install BytePS by yourself. There are two options.
+
+### Build from pip
```
Build from pip --> Install by pip

bobzhuyb commented on README.md:
```diff
@@ -41,21 +46,30 @@ BytePS also incorporates many acceleration techniques such as hierarchical strat
+- BytePS assumes that you have already installed one or more of the following frameworks: TensorFlow / PyTorch / MXNet. 
+- BytePS depends on CUDA and NCCL. You should specify the NCCL path with `export BYTEPS_NCCL_HOME=/path/to/nccl`. By default it points to `/usr/local/nccl`.
+- The installation requires gcc>=4.9. If you are working on CentOS/Redhat and have gcc<4.9, you can try `yum install devtoolset-7` before everything else. In general, we recommend using gcc 4.9 for best compatibility ([example](https://github.com/bytedance/byteps/blob/master/docker/Dockerfile.pytorch#L72-L80)).
+- RDMA support: During setup, the script will automatically detect the RDMA header file. If you want to use RDMA, make sure your RDMA environment has been properly installed and tested before install.
```
Can you be more specific, e.g., on Ubuntu, users need to have xxx..

bobzhuyb commented on docs/best-practice.md:
```diff
@@ -13,15 +13,23 @@ If you have NVLinks, leave `BYTEPS_PCIE_SWITCH_SIZE` unmodified. If you don't kn
+
+When you don't have additional CPU servers, then for each physical machine, you should launch a worker and a server process. We call this *co-locate* mode, and the resource consumption is the same with Horovod (no additional servers). 
+
+If you are using TCP, you will probably get near-identical performance with Horovod-TCP. However, if you are using RDMA, you can set `BYTEPS_ENABLE_IPC=1` to enable the IPC communication between the co-located worker and server. And eventually you will get higher end-to-end performance than Horovod. 
```
Why don't we enable IPC by default, for RDMA?

bobzhuyb commented on CHANGELOG.rst:
```diff
@@ -0,0 +1,16 @@
+* New feature: Enable mixing use of colocate and non-colocate servers, along with a smart tensor allocation strategy.
+* Add support for pip install.
+
+
```
bpslaunch is a new feature, too
